### ISSUE TITLE
Current address scheme should be deprecated

### ISSUE THREAD

**Issue by ripper234** [ethereum/go-ethereum#2059]
(This issue is a protocol layer issue, not a geth issue. I hope this is the right repo for it)

Let's call the current account/address scheme "v1 addresses", e.g. [0x78ff59159fafb4a337fd158fcf8d9603a7c6f03a](https://www.etherchain.org/account/0x78ff59159fafb4a337fd158fcf8d9603a7c6f03a).

The v1 address scheme is faulty because it is [missing a checksum](https://www.reddit.com/r/ethereum/comments/3vxhul/eli5_how_do_a_simple_user_work_around_the_no/), and this can cause users to lose money.

I propose that a new v2 address scheme is designed, that is a "mask" or translation of v1 addresses, in a way that includes a checksum. Eventually, v1 addresses should be completely deprecated from all Ethereum clients, and the clients should only use v2 addresses (all v1 addresses should be invalid addresses when typed into a client).

Initial proposal for v2 addresses:

> v2addr(v1addr) = Concat("E", v1addr, MD5(v1addr)).
> // MD5 and not something stronger because there's no cryptographic need here. Any fast hash will do, it doesn't have to be secure.

There is an easy to compute one to one translation from v1 addresses to v2 and vice versa)

**Comment by obscuren** [comment]
Thanks for the suggestion Ron.

Please see [ICAP](https://github.com/ethereum/wiki/wiki/ICAP:-Inter-exchange-Client-Address-Protocol)

**Comment by ripper234** [comment]
Thanks. I'm closing this now, I guess this isn't the right place for this issue after all.

For future reference, do you think https://github.com/ethereum/wiki/issues is the right place to file issues on the Ethereum protocol?

**Comment by JonnyLatte** [comment]
This wouldn't just be for clients. If a standard address with checksums becomes popular then people are going to want to put these addresses into contract code and have the checksum and version stripped out as the contract compiles.
